hardware_interface/src/resource_manager.cpp

```diff
@@ -699,26 +697,14 @@ class ResourceStorage
     }
   }
 
-  void check_for_duplicates(const HardwareInfo & hardware_info)
-  {
-    // Check for identical names
-    if (hardware_info_map_.find(hardware_info.name) != hardware_info_map_.end())
-    {
-      throw std::runtime_error(
-        "Hardware name " + hardware_info.name +
-        " is duplicated. Please provide a unique 'name' in the URDF.");
-    }
-  }
-
   // TODO(destogl): Propagate "false" up, if happens in initialize_hardware
-  void load_and_initialize_actuator(const HardwareInfo & hardware_info)
+  bool load_and_initialize_actuator(const HardwareInfo & hardware_info)
   {
     auto load_and_init_actuators = [&](auto & container)
     {
-      check_for_duplicates(hardware_info);
```

Comment:
@saikishor I am removing this as we are checking for duplicates in `load_and_initialize_components` that should be a main entry point anyways. This is also how it is used in ros-controls/gz_ros2_control#265.